src/components/CommentsList.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Comment, CommentData } from '../types/Comment';
+import { Loader } from './Loader';
+import { NewCommentForm } from './NewCommentForm';
+import { deleteComment, getComments, postComment } from '../api/comments';
+
+type Props = {
+  postId: number;
+};
+
+export const CommentsList: React.FC<Props> = ({ postId }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadError, setLoadError] = useState(false);
+  const [addError, setAddError] = useState(false);
+  const [deleteError, setDeleteError] = useState(false);
+  const [isFormVisible, setIsFormVisible] = useState(false);
+
+  const getPostCommentsFromServer = async () => {
```

Comment:
Consider renaming the function 'getPostCommentsFromServer' to 'fetchPostComments' for better clarity.